charts/seed-bootstrap/prometheus-rules/recording-rules.rules.yaml

```
@@ -36,6 +36,102 @@ groups:
+    expr: count(max(container_memory_working_set_bytes) by (container, namespace) / max(kube_pod_container_resource_limits_memory_bytes) by (container, namespace) >= 0.8) by (container)
+
+  # Recording rules for the count of containers which have CPU usage at or above 100% of their VPA recommendations, aggregated by containers.
+  - record: seed:containers_over_cpu_recommendation_total:count_by_container
+    expr: count(max(rate(container_cpu_usage_seconds_total[5m])) by (container, namespace) / max(vpa_status_recommendation{recommendation="target", resource="cpu"}) by (container, namespace) >= 1) by (container)
+
+  # Recording rules for the count of containers which have memory usage at or above 100% of their VPA recommendations, aggregated by containers.
+  - record: seed:containers_over_memory_recommendation_total:count_by_container
+    expr: count(max(container_memory_working_set_bytes) by (container, namespace) / max(vpa_status_recommendation{recommendation="target", resource="memory"}) by (container, namespace) >= 1) by (container)
+
+  # Recording rules for the count of containers which have CPU requests below 100% of their VPA recommendations, aggregated by containers.
+  - record: seed:containers_cpu_scale_up_pending_total:count_by_container
+    expr: count(max(kube_pod_container_resource_requests_cpu_cores) by (container, namespace) / max(vpa_status_recommendation{recommendation="target", resource="cpu"}) by (container, namespace) < 1) by (container)
+
+  # Recording rules for the count of containers which have memory requests below 100% of their VPA recommendations, aggregated by containers.
+  - record: seed:containers_memory_scale_up_pending_total:count_by_container
+    expr: count(max(kube_pod_container_resource_requests_memory_bytes) by (container, namespace) / max(vpa_status_recommendation{recommendation="target", resource="memory"}) by (container, namespace) < 1) by (container)
+
+  # Recording rules for the count of containers which have CPU requests above 100% of their VPA recommendations, aggregated by containers.
+  - record: seed:containers_cpu_scale_down_pending_total:count_by_container
+    expr: count(max(kube_pod_container_resource_requests_cpu_cores) by (container, namespace) / max(vpa_status_recommendation{recommendation="target", resource="cpu"}) by (container, namespace) > 1) by (container)
+
+  # Recording rules for the count of containers which have memory requests at or above 100% of their VPA recommendations, aggregated by containers.
+  - record: seed:containers_memory_scale_down_pending_total:count_by_container
+    expr: count(max(kube_pod_container_resource_requests_memory_bytes) by (container, namespace) / max(vpa_status_recommendation{recommendation="target", resource="memory"}) by (container, namespace) > 1) by (container)
+
+  # Recording rules for the minimum CPU usage aggregated by container
+  - record: seed:container_cpu_usage_seconds_total:max_by_container
```

```suggestion
  - record: seed:container_cpu_usage_seconds_total:min_by_container
```

Many recording rules below this are named `*_max_by_container` even though the operator used is `min`. Is this intended?

---

Oops. Missed the substitution. Will fix shortly.